### PR TITLE
Fix ./mage proto:all failure

### LINKED ISSUE #1838: Mage cannot generate protos when folder name is not lorawan-stack
#### Summary

Regenerating the proto files fails when the repository folder is not named `lorawan-stack`

#### Steps to Reproduce

1. `git clone git@github.com:TheThingsNetwork/lorawan-stack custom-lorawan-stack`
2. `cd custom-lorawan-stack`
3. `make mage`
4. `./mage proto:clean proto:all`

#### What do you see now?

```
$ ./mage proto:all
lorawan-stack/api/contact_info.proto: File not found.
lorawan-stack/api/identifiers.proto: File not found.
lorawan-stack/api/rights.proto: File not found.
custom-lorawan-stack/api/application.proto:22:1: Import "lorawan-stack/api/contact_info.proto" was not found or had errors.
custom-lorawan-stack/api/application.proto:23:1: Import "lorawan-stack/api/identifiers.proto" was not found or had errors.
custom-lorawan-stack/api/application.proto:24:1: Import "lorawan-stack/api/rights.proto" was not found or had errors.
...
```

#### What do you want to see instead?

The build to succeed

#### Environment

all

#### How do you propose to implement this?

Update magefile

#### Can you do this yourself and submit a Pull Request?

Can update mage, need review

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #1838 

#### Changes

- Update the mount paths in the docker command and the include path

#### Notes for Reviewers

...

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [X] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [ ] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: .mage/proto.go [hunk 1]
```diff
@@ -69,12 +69,12 @@ func makeProtoc() (func(...string) error, *protocContext, error) {
 	return sh.RunCmd("docker", "run",
 			"--rm",
 			"--user", fmt.Sprintf("%s:%s", usr.Uid, usr.Gid),
-			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/api", filepath.Join(wd, "api"), wd),
+			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/lorawan-stack/api", filepath.Join(wd, "api"), wd),
```
Comment (rvolosatovs) [inline .mage/proto.go]:
```suggestion
			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/lorawan-stack/api", filepath.Join(wd, "api"), filepath.Dir(wd)),
```

Comment (neoaggelos) [reply .mage/proto.go]: Updated, should be OK now

File: .mage/proto.go [hunk 2]
```diff
@@ -69,14 +69,14 @@ func makeProtoc() (func(...string) error, *protocContext, error) {
...
 			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/go.thethings.network/lorawan-stack/pkg/ttnpb", filepath.Join(wd, "pkg", "ttnpb"), protocOut),
-			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/sdk/js", filepath.Join(wd, "sdk", "js"), wd),
-			"-w", wd,
+			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/lorawan-stack/sdk/js", filepath.Join(wd, "sdk", "js"), filepath.Dir(wd)),
+			"-w", filepath.Join(filepath.Dir(wd), "lorawan-stack"),
 			fmt.Sprintf("%s:%s", protocName, protocVersion),
 			fmt.Sprintf("-I%s", filepath.Dir(wd)),
 		), &protocContext{
-			WorkingDirectory: wd,
+			WorkingDirectory: filepath.Dir(wd),
```
Comment (rvolosatovs) [inline .mage/proto.go]:
```suggestion
			WorkingDirectory: mountWD,
```
And then you can remove all the changes below

File: .mage/proto.go [hunk 3]
```diff
@@ -69,14 +69,14 @@ func makeProtoc() (func(...string) error, *protocContext, error) {
 	return sh.RunCmd("docker", "run",
```
Comment (rvolosatovs) [inline .mage/proto.go]: Just define
```go
mountWD := filepath.ToSlash(filepath.Join(filepath.Dir(wd), "lorawan-stack"))
```
above and use it for clarity

`filepath.ToSlash` to support e.g. Windows.

File: .mage/proto.go [hunk 4]
```diff
@@ -69,14 +69,14 @@ func makeProtoc() (func(...string) error, *protocContext, error) {
...
 			"--rm",
 			"--user", fmt.Sprintf("%s:%s", usr.Uid, usr.Gid),
-			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/api", filepath.Join(wd, "api"), wd),
+			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/lorawan-stack/api", filepath.Join(wd, "api"), filepath.Dir(wd)),
```
Comment (rvolosatovs) [inline .mage/proto.go]:
```suggestion
			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/api", filepath.Join(wd, "api"), mountWD),
```

File: .mage/proto.go [hunk 5]
```diff
@@ -69,14 +69,14 @@ func makeProtoc() (func(...string) error, *protocContext, error) {
...
 			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/go.thethings.network/lorawan-stack/pkg/ttnpb", filepath.Join(wd, "pkg", "ttnpb"), protocOut),
-			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/sdk/js", filepath.Join(wd, "sdk", "js"), wd),
-			"-w", wd,
+			"--mount", fmt.Sprintf("type=bind,src=%s,dst=%s/lorawan-stack/sdk/js", filepath.Join(wd, "sdk", "js"), filepath.Dir(wd)),
```
Comment (rvolosatovs) [inline .mage/proto.go]: IDEM

File: .mage/proto.go [hunk 6]
```diff
@@ -69,14 +69,14 @@ func makeProtoc() (func(...string) error, *protocContext, error) {
...
+			"-w", filepath.Join(filepath.Dir(wd), "lorawan-stack"),
```
Comment (rvolosatovs) [inline .mage/proto.go]:
```suggestion
			"-w", mountWD,
```

File: .mage/proto.go [hunk 7]
```diff
@@ -66,17 +66,20 @@ func makeProtoc() (func(...string) error, *protocContext, error) {
 	if err != nil {
 		return nil, nil, xerrors.Errorf("failed to get user: %w", err)
 	}
+
+	mountWD := filepath.ToSlash(filepath.Join(filepath.Dir(wd), "lorawan-stack"))
+
```
Comment (rvolosatovs) [inline .mage/proto.go]: remove newline

**Pull request merged**